### PR TITLE
Scheduler Management endpoint #8600

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

File: modules/server/server-rest/src/main/java/com/enonic/xp/impl/server/rest/SchedulerResource.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    public List<ScheduledJobJson> list()
+    {
+        return schedulerService.list().
+            stream().
+            map( ScheduledJobJson::new ).
+            collect( Collectors.toList() );
+
+    }
+
+    @Reference
```
Comment (rymsha) [inline modules/server/server-rest/src/main/java/com/enonic/xp/impl/server/rest/SchedulerResource.java:40]: Use constructor injection

File: modules/server/server-rest/src/main/java/com/enonic/xp/impl/server/rest/model/ScheduledJobJson.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import com.enonic.xp.scheduler.ScheduleCalendar;
+import com.enonic.xp.scheduler.ScheduledJob;
+
+public class ScheduledJobJson
+{
+    private final ScheduledJob job;
+
+    public ScheduledJobJson( final ScheduledJob job )
+    {
+        this.job = job;
```
Comment (rymsha) [inline modules/server/server-rest/src/main/java/com/enonic/xp/impl/server/rest/model/ScheduledJobJson.java:18]: Copying fields in ctor would look better

File: modules/server/server-rest/src/main/java/com/enonic/xp/impl/server/rest/model/ScheduledJobJson.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    public Instant getLastRun()
+    {
+        return lastRun;
+    }
+
+    public Instant getNextExecution()
+    {
+        return enabled ? calendar.
```
Comment (rymsha) [inline modules/server/server-rest/src/main/java/com/enonic/xp/impl/server/rest/model/ScheduledJobJson.java:93]: move to ShedulerCalendar

File: modules/server/server-rest/src/main/java/com/enonic/xp/impl/server/rest/model/ScheduledJobJson.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    public boolean isEnabled()
+    {
+        return enabled;
+    }
+
+    public String getDescriptor()
+    {
+        return descriptor.toString();
```
Comment (rymsha) [inline modules/server/server-rest/src/main/java/com/enonic/xp/impl/server/rest/model/ScheduledJobJson.java:68]: do not `toString` objects that are String-like

**Review by rymsha — CHANGES_REQUESTED** [review]

**Review by rymsha — APPROVED** [review]

File: modules/server/server-rest/src/main/java/com/enonic/xp/impl/server/rest/SchedulerResource.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import com.enonic.xp.impl.server.rest.model.ScheduledJobJson;
+import com.enonic.xp.jaxrs.JaxRsComponent;
+import com.enonic.xp.scheduler.SchedulerService;
+import com.enonic.xp.security.RoleKeys;
+
+@Path("/scheduler")
```
Comment (anatol-sialitski) [inline modules/server/server-rest/src/main/java/com/enonic/xp/impl/server/rest/SchedulerResource.java:21]: maybe rename to `schedulers`, the GET method list will be available by default and that is right naming for REST

Comment (vbradnitski) [reply modules/server/server-rest/src/main/java/com/enonic/xp/impl/server/rest/SchedulerResource.java:21]: but then we have to rename all exist resources

**Pull request merged**